stdlib/logging/__init__.pyi

```diff
@@ -63,10 +63,13 @@ if sys.version_info >= (3, 11):
 if sys.version_info >= (3, 12):
     __all__ += ["getHandlerByName", "getHandlerNames"]
 
+class _FilterProtocol(Protocol):
+    def filter(self, record: LogRecord) -> bool: ...
+
 _SysExcInfoType: TypeAlias = tuple[type[BaseException], BaseException, TracebackType | None] | tuple[None, None, None]
 _ExcInfoType: TypeAlias = None | bool | _SysExcInfoType | BaseException
 _ArgsType: TypeAlias = tuple[object, ...] | Mapping[str, object]
-_FilterType: TypeAlias = Filter | Callable[[LogRecord], bool]
+_FilterType: TypeAlias = Filter | Callable[[LogRecord], bool] | _FilterProtocol
```

Comment:
Any `Filter` instance has a `.filter()` method, so you can do:

```suggestion
_FilterType: TypeAlias = Callable[[LogRecord], bool] | _FilterProtocol
```

This is IMO slightly simpler / easier to understand.

Reply:
Actually, the explicit `Filter` type is needed after #10540.

Reply:
Let me know how you’d like me to proceed here — keep or remove `Filter` 🤓

Reply:
I'd say remove `Filter` (like you already did), but support returning a `LogRecord` object on 3.12+ just like in #10540.

Reply:
Commit https://github.com/python/typeshed/pull/11018/commits/55e37101cdefa8e7f69b53144745b3b3c0c16a1a.

But then, shouldn’t the `Callable` be adjusted also to return `bool | LogRecord` on Python 3.12+ ?

Reply:
> @Akuli I'd say remove Filter, […]

I guess that broke Sphinx below, although it _does_ inherit from `logging.Filter` [here](https://github.com/sphinx-doc/sphinx/blob/bb74aec2b6aa1179868d83134013450c9ff9d4d6/sphinx/util/logging.py#L473).

Reply:
Somehow I didn't see your last two comments, and commented the same exact things myself :)

Reply:
> Somehow I didn't see your last two comments, and commented the same exact things myself :)

Great minds 👍🏼